### ISSUE TITLE
FutureWarning: Possible nested set at position 7 (in _validators.py line 200) with Python 3.7+

### ISSUE THREAD

**Issue by AdamWill** [Julian/jsonschema#659]
Any time jsonschema is used with Python 3.7+, this warning is printed:

    /usr/lib/python3.8/site-packages/jsonschema/_validators.py:200: FutureWarning: Possible nested set at position 7
      not re.search(patrn, instance)

this seems to be documented [here](https://docs.python.org/dev/whatsnew/3.7.html#re):

"FutureWarning is now emitted if a regular expression contains character set constructs that will change semantically in the future, such as nested sets and set operations. (Contributed by Serhiy Storchaka in bpo-30349.)"

Not sure what's an appropriate fix to avoid this warning for jsonschema, sorry.

**Comment by Julian** [comment]
Thanks. When you say any usage -- can you show a specific piece of code
that produces this for you?

**Comment by AdamWill** [comment]
Well, pretty much any use of [this thing](https://pagure.io/fedora-qa/os-autoinst-distri-fedora/blob/master/f/fifloader.py), is what I'm hitting it with.

**Comment by Julian** [comment]
I don't know what that is, sorry, and it's 350 lines long.

I was asking for some specific code because you mentioned

> Any time jsonschema is used with Python 3.7+, this warning is printed:

but I see no such warning:

```
⊙  python3.8 -m venv venv; venv/bin/python -m pip install jsonschema                                                                                                                              julian@Air
Collecting jsonschema
/Users/julian/Desktop/venv/lib/python3.8/site-packages/pip/_vendor/msgpack/fallback.py:133: DeprecationWarning: encoding is deprecated, Use raw=False instead.
  unpacker = Unpacker(None, max_buffer_size=len(packed), **kwargs)
  Using cached https://files.pythonhosted.org/packages/c5/8f/51e89ce52a085483359217bc72cdbf6e75ee595d5b1d4b5ade40c7e018b8/jsonschema-3.2.0-py2.py3-none-any.whl
Collecting pyrsistent>=0.14.0 (from jsonschema)
Collecting attrs>=17.4.0 (from jsonschema)
  Using cached https://files.pythonhosted.org/packages/a2/db/4313ab3be961f7a763066401fb77f7748373b6094076ae2bda2806988af6/attrs-19.3.0-py2.py3-none-any.whl
Collecting six>=1.11.0 (from jsonschema)
  Using cached https://files.pythonhosted.org/packages/65/eb/1f97cb97bfc2390a276969c6fae16075da282f5058082d4cb10c6c5c1dba/six-1.14.0-py2.py3-none-any.whl
Requirement already satisfied: setuptools in ./venv/lib/python3.8/site-packages (from jsonschema) (41.2.0)
Installing collected packages: six, pyrsistent, attrs, jsonschema
Successfully installed attrs-19.3.0 jsonschema-3.2.0 pyrsistent-0.15.7 six-1.14.0
/Users/julian/Desktop/venv/lib/python3.8/site-packages/pip/_vendor/msgpack/fallback.py:133: DeprecationWarning: encoding is deprecated, Use raw=False instead.
  unpacker = Unpacker(None, max_buffer_size=len(packed), **kwargs)
WARNING: You are using pip version 19.2.3, however version 20.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

~/Desktop
⊙  venv/bin/python -c 'import jsonschema; jsonschema.validate(12, {})'                                                                                                                            julian@Air
```

I'm happy to look at this, but I'll need some short, self-contained code example that you're running -- I can't really look through a long script, it seems more likely than not to be something you're doing there (or whoever wrote that) that's causing the warning but it's hard to tell without reducing the size of it.

**Comment by Julian** [comment]
Happy to take a look / hear more comments if you have more updates here.

**Comment by rubin55** [comment]
Hey! I ran into this + found why this is happening, almost exact same  error message:

```
jsonschema/_validators.py:200: FutureWarning: Possible nested set at position 5
  not re.search(patrn, instance)
```

This happened to me when a `pattern` contained an accidental double `[`. Note that other schema validators, like [this one](https://extendsclass.com/json-schema-validator.html) don't consider this an error and presumably simply add the `[` character to the match set. 

This caused the error:
```
"valid_path": {
  "type": "string",
  "pattern": "^[/][[a-zA-Z0-9/_-]+$"
}
```

The removal of the extra `[` fixed it:
```
"valid_path": {
  "type": "string",
  "pattern": "^[/][a-zA-Z0-9/_-]+$"
}
```
